driver-core/src/main/java/com/datastax/driver/core/Metrics.java

```diff
@@ -262,6 +251,24 @@ void shutdown() {
             jmxReporter.stop();
     }
 
+    private static Gauge<Integer> buildQueueSizeGauge(final ExecutorService executor) {
+        if (executor instanceof ThreadPoolExecutor) {
```

Comment:
I think this doesn't work for executor and blockingExecutor because `listeningDecorator` returns a `ListeningDecorator` which isn't an instance of `ThreadPoolExecutor`.  Noticed that it was returning `-1` for `blocking-executor-queue-depth` and `executor-queue-depth`